microprofile.jdt/org.eclipse.lsp4mp.jdt.core/plugin.xml

```diff
@@ -31,6 +31,8 @@
       <delegateCommandHandler class="org.eclipse.lsp4mp.jdt.internal.core.ls.MicroProfileDelegateCommandHandlerForJava">
             <command id="microprofile/java/fileInfo"/>
             <command id="microprofile/java/codeAction"/>
+            <command id="microprofile/java/codeActionStub"/>
```

Comment:
I wonder if we could skip this codeActionStub by adding in `JavaCodeActionContext ` a method like `isResolveSupport` that MicroProfile LS could fill by using the CodeActionResolveSupportCapabilities. It will give the capability to have just a method getCodeActions and according to `isResolveSupport` the TextEdit should be processed or not at this step.

Reply:
Please let me know if I understood what you are suggesting correctly: we add a parameter `isResolveSupport` to `"microprofile/java/codeAction"`. If `isResolveSupport` is false, we return the resolved CodeActions (like we are currently doing), and if `isResolveSupport` is true, then we return the unresolved CodeActions (everything is filled in except the TextEdit).

If we do it this way, then we still need to contact the JDT component every time CodeActions are requested in the Java component, which if I understand correctly, is what we are trying to avoid and is the motivation behind adding resolve support.